sample/src/main/java/ch/sonect/sdk/shop/integrationapp/MainActivity.kt

```diff
@@ -45,30 +52,53 @@ class MainActivity : AppCompatActivity() {
 
         _merchantId = getDefaultMerchantId()
         etMerchantId.setText(_merchantId)
+        etMerchantId.addTextChangedListener(afterTextChanged {
+            sharedPreferences.merchantId = it.toString()
+        })
 
         _clientId = getClientId()
         etClientId.setText(_clientId)
+        etClientId.addTextChangedListener(afterTextChanged {
+            sharedPreferences.clientId = it.toString()
+        })
 
         _clientSecret = getClientSecret()
         etClientSecret.setText(_clientSecret)
+        etClientSecret.addTextChangedListener(afterTextChanged {
+            sharedPreferences.clientSecret = it.toString()
+        })
 
         _hmacKey = getHmacKey()
         etHmacKey.setText(_hmacKey)
+        etHmacKey.addTextChangedListener(afterTextChanged {
+            sharedPreferences.hmacKey = it.toString()
+        })
+
+        _deviceId = getDeviceId()
+        etDeviceId.setText(_deviceId)
+        etDeviceId.addTextChangedListener(afterTextChanged {
+            sharedPreferences.deviceId = it.toString()
+        })
+
+        when (sharedPreferences.envKey) {
+            "DEV" -> chkDev.isChecked = true
+            "TEST" -> chkTest.isChecked = true
+            "PROD" -> chkProd.isChecked = true
+        }
 
         groupEnviroment.setOnCheckedChangeListener { group, checkedId ->
-            _merchantId = getDefaultMerchantId()
```

Comment:
This will break the logic with different defaults per enviroment.
I'd say we could persist the data and clear all saved stuff when we change enviroment and use env related defaults.

E.g. we have def value AAA for env DEV and def value BBB for TEST

We switch to TEST and value is changed to BBB, change to DEV -> AAA.
Switch again to TEST.

Then change AAA to CCC. Now when you open/close app. Usual default env is DEV but now it's TEST because last used was TEST. And also def value is CCC.

Switch env to DEV. Switch to TEST. Now def value is BBB because it's a default one for TEST.

Reply:
better to not have  any defaults since my defaults might be different from Yogesh. I clear sharedpreferences when switching env.

Reply:
@andre-sonect defaults could be Sonect partner's creds or we could even create 'default' partner and 'default' user inside this partner.

There are 2 main usages of defaults:
- when update SDK in the sample make it easy to run sample to make sure that it least it's not crashing
- Sonect partner's devs could download this sample and run it out of the box. So they could be sure that current code is accurate and it least our own samples works well.